### PR TITLE
MueLu: More alternative coarsening

### PR DESCRIPTION
**Pull request by csiefer2**
Issue: n/a (non-ASC)

Stakeholder Feedback: n/a (non-ASC)

Tests: RHEL7; Added new test to use new code.

Will squash on merge to make @jhux2 happy.

### CODE REVIEW

File: packages/muelu/src/Graph/Containers/MueLu_LWGraph_def.hpp [hunk 1]
```diff
@@ -55,7 +55,7 @@ namespace MueLu {
   //void describe(Teuchos::FancyOStream &out, const VerbLevel verbLevel = Default) const {
   template <class LocalOrdinal, class GlobalOrdinal, class Node>
   void LWGraph<LocalOrdinal, GlobalOrdinal, Node>::print(Teuchos::FancyOStream &out, const VerbLevel verbLevel) const {
-    MUELU_DESCRIBE;
+    //    MUELU_DESCRIBE;
```
Comment (cgcgcg) [inline packages/muelu/src/Graph/Containers/MueLu_LWGraph_def.hpp:58]: No idea what that used to do, but why no more?

File: packages/muelu/src/MueCentral/MueLu_FactoryManager_def.hpp [hunk 2]
```diff
@@ -259,6 +259,8 @@ namespace MueLu {
   void FactoryManager<Scalar, LocalOrdinal, GlobalOrdinal, Node>::Print() const {
     std::map<std::string, RCP<const FactoryBase> >::const_iterator it;
     Teuchos::FancyOStream& fancy = GetOStream(Debug);
+    //auto & fancy = std::cout;// For debugging
```
Comment (cgcgcg) [inline packages/muelu/src/MueCentral/MueLu_FactoryManager_def.hpp:262]: ?

**Review by cgcgcg — APPROVED** [review]

File: packages/muelu/src/Graph/Containers/MueLu_LWGraph_def.hpp (hunk 1 above)
Comment (csiefer2) [reply packages/muelu/src/Graph/Containers/MueLu_LWGraph_def.hpp:58]: That generates a FancyOStream (on rank 0 I think).  We don't want that, as we're passing a specific FOS in.

File: packages/muelu/src/MueCentral/MueLu_FactoryManager_def.hpp (hunk 2 above)
Comment (csiefer2) [reply packages/muelu/src/MueCentral/MueLu_FactoryManager_def.hpp:262]: That was a "I can't figure out how to get the Debug stream to trigger, so I hardcoded cout in instead."

Wanted to leave that in so I remember how to do it again later :)

**Review by cgcgcg — DISMISSED** [review]

**Comment by csiefer2** [comment]
I love it when you can't actually reproduce the test failures...

**Comment by csiefer2** [comment]
@trilinos/framework I cannot reproduce the failing tests following the reproduction instructions.

A little help here?

**Review by cgcgcg — DISMISSED** [review]

**Comment by csiefer2** [comment]
Well that is... random and unrelated

 p=0: *** Caught standard std::exception of type 'std::runtime_error' :
 
  Kokkos failed to allocate memory for label "lgMap".  Allocation using MemorySpace named "Cuda" failed with the following error:  Allocation of size 184 B failed because of an unknown error.  (The allocation mechanism was cudaMalloc().  The Cuda allocation returned the error code ""cudaErrorAssert".)

**Comment by jwillenbring** [comment]
> @trilinos/framework I cannot reproduce the failing tests following the reproduction instructions.
> 
> A little help here?

@csiefer2 What issues are you having? When you try to reproduce on GCC, does everything pass for you? Does it not work generally?

**Comment by csiefer2** [comment]
> @csiefer2 What issues are you having? When you try to reproduce on GCC, does everything pass for you? Does it not work generally?

Got my code sorted out for that error.  The _branch_ was fine, but the branch+develop was not.  Once I rebased the branch on top of develop, I could reproduce the GCC error.

The "reproducing Intel fails doesn't work because Intel won't configure" is a problem that isn't a blocker for me now, but I'd love to see someone take a crack at that.

**Comment by cgcgcg** [comment]
```
Assertion `Kokkos::View ERROR: attempt to access inaccessible memory space` failed.
```

**Review by cgcgcg — DISMISSED** [review]

**Comment by jhux2** [comment]
> Well that is... random and unrelated
> 
> p=0: *** Caught standard std::exception of type 'std::runtime_error' :
> 
> Kokkos failed to allocate memory for label "lgMap". Allocation using MemorySpace named "Cuda" failed with the following error: Allocation of size 184 B failed because of an unknown error. (The allocation mechanism was cudaMalloc(). The Cuda allocation returned the error code ""cudaErrorAssert".)

I saw this yesterday in a different issue, I can't remember which one, though.

**Comment by allevin** [comment]
@csiefer2 @jwillenbring This PR is failing/crashing the Autotester because the last commit https://github.com/trilinos/Trilinos/pull/9321/commits/72760a21a1ba8317723bbcaad5b8e26969123168 has been made by a user that does not have a valid user name (is is called **"Your Name"**).  The AT will continue to stop processing until this commit is corrected

**Comment by csiefer2** [comment]
> 
> 
> @csiefer2 @jwillenbring This PR is failing/crashing the Autotester because the last commit [72760a2](https://github.com/trilinos/Trilinos/commit/72760a21a1ba8317723bbcaad5b8e26969123168) has been made by a user that does not have a valid user name (is is called **"Your Name"**). The AT will continue to stop processing until this commit is corrected

You have to be kidding me.  This is just my luck! ;P

**Review by cgcgcg — APPROVED** [review]

**Review by cgcgcg — APPROVED** [review]

**Pull request merged**